test/unit/abilities/provider_any_test.rb

```diff
@@ -3,15 +3,25 @@
 require 'test_helper'
 
 module Abilities
-  class ProviderAnyTest < ActiveSupport::TestCase
+
+  class BaseTest < ActiveSupport::TestCase
     setup do
-      @account = FactoryBot.create(:simple_provider)
+      @account = FactoryBot.create(:provider_account)
       @account.stubs(:provider_can_use?).with(any_parameters).returns(false)
       @user  = FactoryBot.create(:simple_user, account: @account)
     end
 
     attr_reader :user, :account
 
+    private
+
+    def ability
+      Ability.new(user)
```

Comment:
Why not memoize?
```suggestion
      @ability ||= Ability.new(user)
```

Reply:
Because it makes some test fail, for instance:

https://github.com/3scale/porta/blob/0e2d18a735c06e748e666e3763c6d109a485b543/test/unit/abilities/provider_any_test.rb#L81

This fails because we change user member permissions on the fly but the ability is not updated because it's returning the memoized object.